test/test_cartesian_power.cpp

```diff
@@ -0,0 +1,301 @@
+
+// Copyright (c) 2022 Tristan Brindle (tcbrindle at gmail dot com)
+// Copyright (c) 2023 NVIDIA Corporation (reply-to: brycelelbach@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include "catch.hpp"
+
+#include <array>
+
+#include <flux/core.hpp>
+#include <flux/core/numeric.hpp>
+
+#include "test_utils.hpp"
+
+/*
+ * Reference implementation of repeating version of cartesian_product.
+ */
+namespace flux {
+namespace detail {
+
+template<int RepeatCount>
+struct cartesian_product_repeat_n_fn {
+  template<adaptable_sequence Seq, std::size_t... Indices>
+  requires (multipass_sequence<Seq>)
+  [[nodiscard]]
+  constexpr auto impl(Seq &&seq, std::index_sequence<Indices...>) const {
+    return cartesian_product_adaptor<std::decay_t<decltype((static_cast<void>(Indices),
+        seq))>...>(
+        (static_cast<void>(Indices), seq)...);
+  }
+
+  template<adaptable_sequence Seq>
+  requires (multipass_sequence<Seq>)
+  [[nodiscard]]
+  constexpr auto operator()(Seq &&seq) const {
+    return impl(FLUX_FWD(seq), std::make_index_sequence<RepeatCount>{});
+  }
+};
+
+}
+
+template<int RepeatCount>
+FLUX_EXPORT inline constexpr auto cartesian_product_repeat_n = detail::cartesian_product_repeat_n_fn<RepeatCount>{};
+}
```

Comment:
This is really testing the existing `cartesian_product` rather than the new `cartesian_power`, so I don't think it's necessary here